File: website/Entities/BookCopy.cs
using System;

namespace website.Entities
{
  public class BookCopy
  {
    public string BarCode { get; set; }
    public bool? IsLost { get; set; }
    public DateTime? LostDate { get; set; }
    public bool? IsDamaged { get; set; }
    public DateTime? DamagedDate { get; set; }
  }
}

File: HomeReadingLibraryWeb/ClientApp/src/app/entities/book-copy.ts
export class BookCopy {
    barCode: string;
    isLost?: boolean;
    lostDate: Date;
    isDamaged?: boolean;
    damagedDate: Date;

    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    static fromObject(bookCopyObj: any): BookCopy {
        const bookCopy = new BookCopy(bookCopyObj.barCode);
        Object.assign(bookCopy, bookCopyObj);
        return bookCopy;
    }

    /**
     *
     */
    constructor(barCode: string) {
        this.barCode = barCode;
    }
}
